pkg/controller/elasticsearch/remotecluster/annotations.go

```diff
@@ -33,14 +33,32 @@ func getRemoteClustersInAnnotation(es esv1.Elasticsearch) map[string]struct{} {
 }
 
 func annotateWithCreatedRemoteClusters(c k8s.Client, es esv1.Elasticsearch, remoteClusters map[string]struct{}) error {
+	if len(remoteClusters) == 0 {
+		// if there are no annotations, there's nothing to do
+		if len(es.Annotations) == 0 {
+			return nil
+		}
+
+		// if the annotation exists, delete it
+		if _, ok := es.Annotations[ManagedRemoteClustersAnnotationName]; ok {
+			delete(es.Annotations, ManagedRemoteClustersAnnotationName)
+			return c.Update(&es)
```

Comment:
nit: I'm wondering if it's worth it to cover this part of the code with a unit test (the current unit tests does not check that the annotation has been actually removed)